register/src/register.rs

```diff
@@ -0,0 +1,366 @@
+use clap::{
+    crate_description, crate_name, crate_version, App, AppSettings, Arg, ArgMatches, SubCommand,
+};
+use serde_derive::Serialize;
+use solana_client::rpc_client::RpcClient;
+use solana_config_api::{config_instruction, ConfigState};
+use solana_sdk::message::Message;
+use solana_sdk::signature::{read_keypair, Keypair, KeypairUtil};
+use solana_sdk::transaction::Transaction;
+use std::error;
+use std::process::exit;
+
+pub const MAX_SHORT_FIELD_LENGTH: usize = 70;
+pub const MAX_LONG_FIELD_LENGTH: usize = 256;
+pub const JSON_RPC_URL: &str = "https://api.testnet.solana.com/";
+
+// Config account key: Va1idator1nfo111111111111111111111111111111
+pub const REGISTER_CONFIG_KEY: [u8; 32] = [
+    7, 81, 151, 1, 116, 72, 242, 172, 93, 194, 60, 158, 188, 122, 199, 140, 10, 39, 37, 122, 198,
+    20, 69, 141, 224, 164, 241, 111, 128, 0, 0, 0,
+];
+
+solana_sdk::solana_name_id!(
+    REGISTER_CONFIG_KEY,
+    "Va1idator1nfo111111111111111111111111111111"
+);
+
+#[derive(Debug, Serialize)]
+struct ValidatorInfo {
+    #[serde(with = "key_value_vec")]
+    info: Vec<String>,
+}
+
+mod key_value_vec {
+    use serde::ser::{self, SerializeTuple, Serializer};
+
+    pub fn serialize<S: Serializer>(elements: &[String], serializer: S) -> Result<S::Ok, S::Error> {
+        // Pass a non-zero value to serialize_tuple() so that serde_json will
+        // generate an open bracket.
+        let mut seq = serializer.serialize_tuple(1)?;
+
+        for element in elements {
+            let len = element.len();
+            if len > std::u8::MAX as usize {
+                return Err(ser::Error::custom("length larger than u8"));
+            }
+            seq.serialize_element(&(len as u8))?;
+            for byte in element.as_bytes() {
+                seq.serialize_element(&byte)?;
+            }
+        }
+        seq.end()
+    }
+}
+
+impl ConfigState for ValidatorInfo {
+    fn max_space() -> u64 {
+        512
```

Comment:
6 lines of text...

Reply:
@rob-solana , are you suggesting that's too much or too little? Or...?

Reply:
seems small to me